src/compiler/checker.ts

```diff
@@ -22628,11 +22628,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             let i = 0;
             for (const type of target.types) {
                 const targetType = getTypeOfPropertyOfType(type, propertyName);

```

Comment:
Can this calculation be inside the loop so we aren't doing any work if the i-th element has already been calculated?

Reply:
Good point.

Reply:
Though, it could be a behavior change... Doubt it, though.

Reply:
If it changes stuff, then I truly know nothing about booleans and loops lol.

Reply:
Usually it comes down to "will this call check something that causes an error that hasn't previously been seen before?".